tutorials/3.Interactions.md

```diff
@@ -0,0 +1,348 @@
+
+### Closing off and caveats
+
+There are some more complex constructions with slash commands (such as subcommands), but as they are currently a bit fiddly we won't cover them in this tutorial.
+
+Discord also offers user and message application commands, which we haven't created nice interfaces for just yet, but are usable if you do create them.
+
+## Components
+
+Another cool thing added was a variety of widgets and gizmos that bots can add to messages called "components". These also use the interaction system to process, but we've abstracted that again.
+
+First, let's decide what we want to do with this. How about we add to the basic `myping`, and make it so that there's a button that says "Ping!", which people can click and it'll reply "pong"?
+
+Yes it's contrived, I'm sorry.
+
+Like before, this is a two stage process. In one place, we have to create the component itself, and in another we have to handle the interaction the component generates. 
+
+Let's create the component itself. We need to add it to the button to be pressed. This isn't too streamlined, but it makes some level of sense.
+
+A button needs the text it will display, and a unique identifier that we'll use to differentiate and process a button. In this case, those are "Ping!" and "myping pingbutton" (why that exactly I'll explain later).
+
+```haskell
+myPingAction :: DatabaseDiscord MessageDetails
+myPingAction = return $ (messageDetailsBasic "pong") { messageDetailsComponents = Just [cps] }
+    where cps = ActionRowButtons [mkButton "Ping!" "myping pingbutton"]
+```
+
+Running the bot, we see that we get the button now! But it just loads for a couple seconds then errors. We need our bot to actually handle this button press.
+
+We can construct and add this component handler by using `processComponentInteraction` and `ComponentRecv`. The first creates a parser like `parseComm` did in the [Plugins](./1.Plugins.md) tutorial and the latter creates the data structure this interaction processor works in.
+
+The action we use in response to the button will be `myPingAction` as it was before, so we feed that and `False` to `processComponentInteraction`. The `False` means that we send a message, instead of updating a message (see [More complex components](#more-complex-components)).
+
+The reason I chose "myping pingbutton" as the identifier before is because of how we process and distribute components. The first word of the unique identifier has to be the plugin name, and the second has to be identifier of the particular component being processed.
+
+```haskell
+myPingButton :: ComponentRecv
+myPingButton = ComponentRecv "pingbutton" (processComponentInteraction myPingAction False)
+```
+
+Now we just load `myPingButton` into `onComponentRecvs` in our plugin creation, run the bot again, run the command, and press our sparkly button, and we get a message from the bot saying "pong"!
```

Comment:
Please provide an example of this - I'm not exactly sure what you're referring to here.